### PR TITLE
Docs and yaml tests for viewing API key's limited-by

### PR DESCRIPTION
**Pull request by ywangd**
This PR updates relevant docs and yaml tests to cover the new feature
of viewing API key's limited-by role descriptors introduced in #89273

Relates: #89058

### CODE REVIEW

File: rest-api-spec/src/main/resources/rest-api-spec/api/security.get_api_key.json [hunk 1]
```diff
@@ -40,6 +40,11 @@
         "type":"boolean",
         "default":false,
         "description":"flag to query API keys owned by the currently authenticated user"
+      },
+      "with_limited_by":{
+        "type":"boolean",
+        "default":false,
+        "description": "flag to show the limited-by role descriptors of API Keys"
       }
```
Comment (ywangd) [inline rest-api-spec/src/main/resources/rest-api-spec/api/security.get_api_key.json:48]: @elastic/clients-team This PR adds a new query parameter to both GetApiKey and QueryApiKey APIs.

File: x-pack/docs/en/rest-api/security/get-api-keys.asciidoc [hunk 2]
```diff
@@ -55,6 +55,13 @@ by the currently authenticated user. Defaults to false.
 The 'realm_name' or 'username' parameters cannot be specified when this
 parameter is set to 'true' as they are assumed to be the currently authenticated ones.
 
+`with_limited_by`::
+(Optional, Boolean) A boolean flag to return the snapshot of the owner user's role descriptors
+associated with the API key. An API key's actual permission is the intersection of
+its <<api-key-role-descriptors,assigned role descriptors>> and the owner user's role descriptors
+(effectively limited by it). An API key cannot retrieve any API key's limited-by role descriptors
+(including itself) unless it has `manage_api_key` or higher privileges.
```
Comment (n1v0lg) [inline x-pack/docs/en/rest-api/security/get-api-keys.asciidoc:63]: Optional:

```suggestion
(effectively limited by it). An API key must have `manage_api_key` or higher privileges to retrieve the limited-by role descriptors of any API key, including itself.
```

File: x-pack/docs/en/rest-api/security/get-api-keys.asciidoc [hunk 3]
```diff
@@ -94,11 +104,80 @@ You can use the following example to retrieve the API key by ID:
...
+            }
+          }
+        }
+      ]
+    }
+  ]
+}
+--------------------------------------------------
+// NOTCONSOLE
+<1> The list of API keys that were retrieved for this request.
```
Comment (n1v0lg) [inline x-pack/docs/en/rest-api/security/get-api-keys.asciidoc:161]: Nit preference but I would omit notes on the obvious stuff, like `id` and `name` -- it mostly just clutters the doc and it's harder to find the important info IMO.

Comment (n1v0lg) [reply x-pack/docs/en/rest-api/security/get-api-keys.asciidoc:161]: Ah I see, it's a copy from below. Good with keeping as is, in that case.

**Review by n1v0lg — APPROVED** [review]
LGTM

**Comment by ywangd** [comment]
@elasticmachine update branch

**Review by sethmlarson — APPROVED** [review]
LGTM

**Review by lockewritesdocs — APPROVED** [review]
LGTM 🦖

**Pull request merged**